Source/AwsCommonRuntimeKit/mqtt/Mqtt5Client.swift

```diff
@@ -121,44 +121,154 @@ public typealias OnWebSocketHandshakeInterceptComplete = (HTTPRequestBase, Int32
+    public func stop(disconnectPacket: DisconnectPacket? = nil) throws {
+        try self.clientCore.stop(disconnectPacket: disconnectPacket)
+    }
+
+    /// Tells the client to attempt to subscribe to one or more topic filters.
+    ///
+    /// - Parameters:
+    ///     - subscribePacket: SUBSCRIBE packet to send to the server
+    /// - Returns:
+    ///     - `SubackPacket`: return Suback packet if the subscription operation succeeded
+    ///
+    /// - Throws: CommonRuntimeError.crtError
+    public func subscribe(subscribePacket: SubscribePacket) async throws -> SubackPacket {
+        return try await clientCore.subscribe(subscribePacket: subscribePacket)
+    }
+
+    /// Tells the client to attempt to publish to topic filter.
+    ///
+    /// - Parameters:
+    ///     - publishPacket: PUBLISH packet to send to the server
+    /// - Returns:
+    ///     - For qos 0 packet: return `None` if publish succeeded
+    ///     - For qos 1 packet: return `PublishResult` packet if the publish succeeded
+    ///
+    /// - Throws: CommonRuntimeError.crtError
+    public func publish(publishPacket: PublishPacket) async throws -> PublishResult {
+        return try await clientCore.publish(publishPacket: publishPacket)
+    }
+
+    /// Tells the client to attempt to unsubscribe to one or more topic filters.
+    ///
+    /// - Parameters:
+    ///     - unsubscribePacket: UNSUBSCRIBE packet to send to the server
+    /// - Returns:
+    ///     - `UnsubackPacket`: return Unsuback packet if the unsubscribe operation succeeded
+    ///
+    /// - Throws: CommonRuntimeError.crtError
+    public func unsubscribe(unsubscribePacket: UnsubscribePacket) async throws -> UnsubackPacket {
+        return try await clientCore.unsubscribe(unsubscribePacket: unsubscribePacket)
+    }
+
+    /// Force the client to discard all operations and cleanup the client.
+    public func close() {
+        clientCore.close()
+    }
+
+    deinit {
+        clientCore.close()
+    }
+
+}
+
+// MARK: - Internal/Private
+
+/// Mqtt5 Client Core, internal class to handle Mqtt5 Client operations
+public class Mqtt5ClientCore {
+    internal var rawValue: UnsafeMutablePointer<aws_mqtt5_client>?
+    internal let rwlock = ReadWriteLock()
+
+    ///////////////////////////////////////
+    // user callbacks
+    ///////////////////////////////////////
+    internal let onPublishReceivedCallback: OnPublishReceived
+    internal let onLifecycleEventStoppedCallback: OnLifecycleEventStopped
+    internal let onLifecycleEventAttemptingConnect: OnLifecycleEventAttemptingConnect
+    internal let onLifecycleEventConnectionSuccess: OnLifecycleEventConnectionSuccess
+    internal let onLifecycleEventConnectionFailure: OnLifecycleEventConnectionFailure
+    internal let onLifecycleEventDisconnection: OnLifecycleEventDisconnection
+    // The websocket interceptor could be nil if the websocket is not in use
+    internal let onWebsocketInterceptor: OnWebSocketHandshakeIntercept?
```

Comment:
Debatable: I feel like internal is redundant, I get the idea that it is more explicit. However, marking stuff internal explicitly does beg the question of why internal and not private?

Reply:
I set the member functions to internal so that we can directly access the member function in the callbacks function:
For example: `internal func MqttClientHandleLifecycleEvent(_ lifecycleEvent: UnsafePointer<aws_mqtt5_client_lifecycle_event>?) `
https://github.com/awslabs/aws-crt-swift/blob/8867080300bf91f2306070dd26b71c273990aa04/Source/AwsCommonRuntimeKit/mqtt/Mqtt5Client.swift#L447

Reply:
Should this be fileprivate then? I am fine with internal, it's just if you explicitly say it, it makes me wonder why internal. If you don't state it, it's more like I don't want it to be public, just give me default internal scope.